CardForm/src/main/java/com/braintreepayments/cardform/view/CvvEditText.java

```diff
@@ -65,10 +69,15 @@ public void setCardType(CardType cardType) {
      * @param mask if {@code true}, this field will be masked.
      */
     public void setMask(boolean mask) {
+        if (mDefaultTransformationMethod == null) {
+            // capture reference to allow fallback to default TextView behavior
+            mDefaultTransformationMethod = getTransformationMethod();
+        }
+
         if (mask) {
-            setInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_VARIATION_PASSWORD);
+            setTransformationMethod(PasswordTransformationMethod.getInstance());
         } else {
-            setInputType(InputType.TYPE_CLASS_NUMBER);
+            setTransformationMethod(mDefaultTransformationMethod);
```

Comment:
Since there aren't any getters/setters for `mDefaultTransformationMethod`, I don't really understand its purpose here. If `mask` is false, should we just set the transformation method to null?

Reply:
Or based on the tests, I guess if `mask` is false, the transformation method should be `SingleLineTransformationMethod`?

Reply:
Yeah technically `SingleLineTransformationMethod` is the default TextViewTransformation.

If Android decides to change this default behavior in the future the test will break, but the actual code will fall back to whatever Android uses by default for TextViews.

Reply:
We'll revert this the behavior probably won't change tbh.

And I agree makes more sense to better align the implementation with the test assertions.